CaesarCipher/CaesarCipher/main.m

```diff
@@ -50,11 +51,149 @@ - (NSString *)decode:(NSString *)string offset:(int)offset {
     return [self encode:string offset: (26 - offset)];
 }
 
+    // check two strings before decoding them.
+    // If they complete these first sonditions, then decode, and compare if there is the same input message
+BOOL checkTwoStrings (NSString *S1, NSString *S2, int off1, int off2) {
+    
+    // check string's length, If they aren't the same length
+    // then we don't need to loop
+    NSInteger count = [S1 length];
+    if (count != [S2 length]) {
+        NSLog(@"Strings don't have the same length");
+        return YES;
+    }
+    //if strings are the same, but with different offsets
+    else if (S1 == S2 && off1!=off2) {
+        NSLog(@"Not the same input message! ");
+        return YES;
+    }
+    // if strings different, but with the same offset
+    else if (S1 != S2 && off1==off2) {
+        NSLog(@"Not the same input message! ");
+        return YES;
+    }
+    // if same strings, and same offset
+    else if (S1 == S2 && off1==off2) {
+        NSLog(@"You entered same strings, and same offsets! ");
+        return YES;
+    }
+    return 0;
+}
+
+    // use codeBreaker only after decoding two trings
+- (BOOL) codeBreaker: (NSString *)string1 compareWith: (NSString *)string2{
+    for (int i = 1; i <26; i++) {
+    //if decoded strings are equal
+            if ([[self encode: string1 offset:i] isEqualToString:[self encode:string2 offset:i]]) {
+                NSLog(@"Same input message!");
+                return YES;
+                break;
+            }
+    // if decoded strings are different
+            else if ([[self encode: string1 offset:i] isNotEqualTo:[self encode:string2 offset:i]]) {
+                    NSLog(@"Not the same input message!");
+                return NO;
+                break;
+            }
+    }
+    return 0;
+}
```

Comment:
This loop will only ever execute 1 time at the most. In either situation you are calling `return` AND `break`. Since the 2 conditions you are checking are opposite, 1 will always be true.